Doc/library/sqlite3.rst

```diff
@@ -2358,6 +2330,84 @@ can be found in the `SQLite URI documentation`_.
 .. _SQLite URI documentation: https://www.sqlite.org/uri.html
 
 
+.. _sqlite3-row-factory-how-to:
+
+How to work with row factories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, :mod:`!sqlite3` represent fetched rows as :class:`tuples <tuple>`.
+If a :class:`!tuple` does not suit your needs,
+use the built-in :class:`Row` type or a custom :attr:`~Connection.row_factory`.
+The former provides both indexed and case-insensitive named access to columns,
+with low memory overhead and minimal performance impact.
+Example use:
+
+.. doctest::
+
+   >>> con = sqlite3.connect(":memory:")
+   >>> con.row_factory = sqlite3.Row
+   >>> res = con.execute("SELECT 'Earth' AS name, 6378 AS radius")
+   >>> row = res.fetchone()
+   >>> row.keys()
+   ['name', 'radius']
+   >>> row[0], row["name"]  # Access by index and name.
+   ('Earth', 'Earth')
+   >>> row["RADIUS"]  # Column names are case-insensitive.
+   6378
+
+If you need more flexibility, you can design your own row factory.
+Here's an example of a :class:`dict` row factory:
+
+.. doctest::
+
+   >>> def dict_factory(cursor, row):
+   ...     col_names = [col[0] for col in cursor.description]
+   ...     return {key: value for key, value in zip(col_names, row)}
+
+   >>> con = sqlite3.connect(":memory:")
+   >>> con.row_factory = dict_factory
+   >>> for row in con.execute("SELECT 1 AS a, 2 AS b"):
+   ...     print(row)
+   {'a': 1, 'b': 2}
+
+Here's an example of a optimised :class:`~collections.namedtuple` factory:
+
+.. testcode::
+
+   from collections import namedtuple
+   from functools import lru_cache
+
+   def _fields(cursor):
+        return (col[0] for col in cursor.description)
+
+   @lru_cache
```

Comment:
IMO, it would be helpful for newer users to have a comment here briefly summarizing how this optimization works, if its not described in the introductory prose.

Reply:
That's a good point, and I think we should go a little bit further: this example does not need to be optimised; it should be to the point. It is better to add the optimisation afterwards, perhaps just as prose. Something like...:

_"As an optimisation, decorate `_make_cls` with `functools.lru_cache` to avoid creating named tuples with the same column spec. See `functools.lru_cache` for more details."_

Reply:
(FYI, I haven't added the above yet.)

Reply:
FYI, I've added a sentence about this in the last commit.